pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -341,39 +343,79 @@ func (o *Options) Run() error {
-				fmt.Fprintf(w, "%s\t%s\n", update.Version, update.Image)
+				fmt.Fprintf(w, "  %s\t%s\n", update.Version, update.Image)
 			}
 			w.Flush()
-			if c := findCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
+			if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 				fmt.Fprintf(o.ErrOut, "warning: Cannot refresh available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
 			}
 		} else {
-			if c := findCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
+			if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 				fmt.Fprintf(o.ErrOut, "warning: Cannot display available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
 			} else {
 				fmt.Fprintf(o.Out, "No updates available. You may force an upgrade to a specific release image, but doing so may not be supported and may result in downtime or data loss.\n")
 			}
 		}
 
+		if o.IncludeNotRecommended {
+			if containsNotRecommendedUpdate(cv.Status.ConditionalUpdates) {
+				sortConditionalUpdatesBySemanticVersions(cv.Status.ConditionalUpdates)
+				fmt.Fprintf(o.Out, "\nSupported but not recommended updates:\n\n")
+				for _, update := range cv.Status.ConditionalUpdates {
+					printNotRecommendedUpdate(o, update)
+				}
+			} else {
+				fmt.Fprintf(o.Out, "\nNo updates which are not recommended based on your cluster configuration are available.\n")
+			}
+		} else if containsNotRecommendedUpdate(cv.Status.ConditionalUpdates) {
+			fmt.Fprintf(o.Out, "\nAdditional updates which are not recommended based on your cluster configuration are available, to view those re-run the command with --include-not-recommended.\n")
+		}
+
 		// TODO: print previous versions
 	}
 
 	return nil
 }
 
+func containsNotRecommendedUpdate(updates []configv1.ConditionalUpdate) bool {
+	for _, update := range updates {
+		if c := findCondition(update.Conditions, "Recommended"); c != nil && c.Status != metav1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func printNotRecommendedUpdate(o *Options, update configv1.ConditionalUpdate) {
+	c := findCondition(update.Conditions, "Recommended")
+	if c == nil || c.Status == metav1.ConditionTrue {
+		return
+	}
+
+	fmt.Fprintf(o.Out, "  Version: %s\n  Image: %s\n", update.Release.Version, update.Release.Image)
+	fmt.Fprintf(o.Out, "  Recommended: %s\n  Reason: %s\n", c.Status, c.Reason)
+	if c.Reason == "MultipleReasons" {
+		indented_message := "    " + strings.Replace(c.Message, "\n", "\n    ", -1)
+		fmt.Fprintf(o.Out, "  Message:\n%s\n", indented_message)
+	} else {
+		fmt.Fprintf(o.Out, "  Message: %s\n", c.Message)
```

Comment:
`MultipleReasons` is more likely to be multi-line, but even single-reason messages could be multi-line.  Can we use `strings.ReplaceAll(c.Message, "\n", "\n    ")`, to follow the pattern from #952, regardless of `Reason`?

Reply:
Yes, it's definitely better to follow the same style.

Since the previous function `printNotRecommendedUpdate` would now have no logic, I have moved its modified body to the place of its original calling.  Every other printing is in the function `Run` also. Let me know if I should change that.